### PR TITLE
Create rabbitmq exchange if not existing

### PR DESCRIPTION
**Pull request by ssd04**
- when starting the rabbit publisher service, verify if the rabbtitMQ exchanges from config are already created, and if not create them

### CODE REVIEW

File: rabbitmq/rabbitClient.go [hunk 1]
```diff
@@ -39,6 +39,19 @@ func NewRabbitMQClient(url string) (*rabbitMqClient, error) {
 	return rc, nil
 }
 
+// ExchangeDeclare will check and declare an exchange if it's not existing already
```
Comment (sstanculeanu) [inline rabbitmq/rabbitClient.go:42]:
```suggestion
// ExchangeDeclare will declare an exchange
```
?

**Review by sstanculeanu — DISMISSED** [review]

File: rabbitmq/publisher.go [hunk 2]
```diff
@@ -54,29 +53,65 @@ func NewRabbitMqPublisher(args ArgsRabbitMqPublisher) (*rabbitMqPublisher, error
...
 }
 
 func checkArgs(args ArgsRabbitMqPublisher) error {
 	if check.IfNil(args.Client) {
 		return ErrNilRabbitMqClient
 	}
-	if args.Config.EventsExchange == "" {
-		return fmt.Errorf("empty events exchange name: %w", ErrInvalidRabbitMqExchangeName)
+
+	return nil
```
Comment (iulianpascalau) [inline rabbitmq/publisher.go:71]: please refactor: call in this function the checkAndCreateExchanges function (it doesn't need to be part of the rabbitMqPublisher struct).

File: rabbitmq/publisher.go [hunk 3]
```diff
@@ -54,29 +53,65 @@ func NewRabbitMqPublisher(args ArgsRabbitMqPublisher) (*rabbitMqPublisher, error
...
+	}
+	err = rp.checkAndCreateExchange(rp.cfg.BlockScrsExchange)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (rp *rabbitMqPublisher) checkAndCreateExchange(conf config.RabbitMQExchangeConfig) error {
```
Comment (iulianpascalau) [inline rabbitmq/publisher.go:100]: this function does 2 things: split maybe?

File: rabbitmq/rabbitClient.go [hunk 4]
```diff
@@ -39,6 +39,19 @@ func NewRabbitMQClient(url string) (*rabbitMqClient, error) {
 	return rc, nil
 }
 
+// ExchangeDeclare will declare an exchange
+func (rc *rabbitMqClient) ExchangeDeclare(name, kind string) error {
+	return rc.ch.ExchangeDeclare(
+		name,  // name
+		kind,  // type
+		true,  // durable
```
Comment (iulianpascalau) [inline rabbitmq/rabbitClient.go:47]: please remove the comments and add variables for L47-L51

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: rabbitmq/publisher.go (hunk 2 above)
Comment (ssd04) [reply rabbitmq/publisher.go:102]: separated args check from exchanges creation

File: rabbitmq/publisher.go (hunk 3 above)
Comment (ssd04) [reply rabbitmq/publisher.go]: splitted check and create

File: rabbitmq/rabbitClient.go (hunk 4 above)
Comment (ssd04) [reply rabbitmq/rabbitClient.go]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by liviuf-ancas — APPROVED** [review]
tested and looks good

**Pull request merged**